libcrux-ml-dsa/src/hash_functions.rs

```diff
@@ -20,12 +40,35 @@ pub(crate) mod shake128 {
 
     pub(crate) trait Xof {
         fn init_absorb(input: &[u8]) -> Self;
+
         // TODO:
         // - There should only be a `squeeze_block` and `squeeze_five_blocks`
         // - Use mutable out instead if performance is better
         fn squeeze_first_five_blocks(&mut self) -> [u8; FIVE_BLOCKS_SIZE];
```

Comment:
I think it is worth checking if mutable out for 5 blocks may provide better perf.

Reply:
Yeah I think that's worth doing eventually because it's big enough again.